TESTING.md

```diff
@@ -0,0 +1,111 @@
+# Testing
+
+The test suite uses [Docker](https://www.docker.com/community-edition) to spin up the following databases when running `npm test`:
+
+| Database      | Service Name |
+|---------------|--------------|
+| PostgreSQL    | `postgres` |
+| MySQL         | `mysql` |
+| MongoDB       | `mongodb` |
+| Elasticsearch | `elasticsearch` |
+| Cassandra     | `cassandra` |
+| SQL Server    | `mssql` |
+| Redis         | `redis` |
+
+Note:
+The `npm test` command uses [Docker Compose](https://docs.docker.com/compose/install/).
+If you have installed Docker,
+you most likely have Docker Compose as well.
+
+## Run tests inside Docker
+
+You can also run the test site it self inside of a Docker container:
+
+```
+npm test -- --docker [node_version] [packages]
+```
+
+Arguments:
+
+- `node_version` - Specify major version of Node.js to run test suite on (default: same version as is installed locally)
+- `packages` - Comma separated list of npm package names for which to run [tav](https://github.com/watson/test-all-versions) tests.
+  If used,
+  `node_version` must be specified (default: none)
+
+Example running test suite on Node.js 8:
+
+```
+npm test -- --docker 8
+```
+
+Example running PostgreSQL and Redis tav tests on Node.js 8:
+
+```
+npm test -- --docker 8 pg,redis
+```
+
+## Run tests outside Docker
+
+If running the test suite multiple times,
+it might be convenient to start these these databases in advance so that `npm test` doesn't have to do that.
+There's two ways to do this:
+
+### Run databases via Docker
+
+Start all databases:
+
+```
+npm run docker-start
+```
+
+Start a subset of databases:
+
+```
+npm run docker-start mssql cassandra
+```
+
+### Run databases locally
+
+If you already have _all_ the required databases installed and running locally,
+just run `npm test` with the `--no-deps` flag:
+
+```
+npm test -- --no-deps
+```
+
+If you only have a subset of the required databases installed and running locally,
+run `npm test` with a list of the datases you want it to start for you,
+e.g:
+
+```
+npm test mssql cassandra
```

Comment:
Shouldn't this be `npm test -- mssql cassandra`?

Reply:
No, the only reason why the `--` is needed in the other commands is because `--docker` and `--no-deps` starts with `--` it self. Then npm thinks that it's a config option for npm and will not pass it on to the script. Therefore I was actually thinking about changing `--docker` to `all` and `--no-deps` to just `none`. That way you'd not have to remember to add the extra `--`. What do you think?

Reply:
Ah, I didn't know npm did that. 😅 

I like the idea of using `all` and `none` instead of the flags.

Reply:
Updated to use `all` and `none`